pkg/tfbridge/provider.go

```diff
@@ -765,6 +770,42 @@ func (p *Provider) Check(ctx context.Context, req *pulumirpc.CheckRequest) (*pul
 		return nil, err
 	}
 
+	logger := GetLogger(ctx)
+	// for now we are just going to log warnings if there are failures.
+	// over time we may want to turn these into actual errors
+	_, validateShouldError := os.LookupEnv("PULUMI_ERROR_TYPE_CHECKER")
+	schemaMap, schemaInfos := res.TF.Schema(), res.Schema.GetFields()
+	if p.pulumiSchema != nil {
+		schema := sync.OnceValue(func() *pschema.PackageSpec {
```

Comment:
Hm this is not going to do any caching I think since the state created by sync.OnceValue(f) is not stored in p.Provider therefore it's going to be re-created and GC-d on every call.

Reply:
@t0yv0 I moved this sync.OnceValue function to the `Provider` object. I _think_ this does what we are wanting, but I'm not sure if we really need the `OnceValue` function at all. Is there a downside to unmarshaling it inside `newProvider` and just storing it on the `Provider` object?

Reply:
>  Is there a downside to unmarshaling it inside newProvider and just storing it on the Provider object?

No downside, that'd be the more natural thing to do.